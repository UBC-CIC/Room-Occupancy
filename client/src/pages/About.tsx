import * as React from "react";
import { Header } from "semantic-ui-react";
import "../App.css";

import Woody from "../assets/images/team/woody.png";
import Colin from "/assets/images/team/colin.jpg";
import Steve from "/assets/images/team/steve.jpg";
import Cassiel from "/assets/images/team/cassiel.png";
import Danny from "/assets/images/team/danny.png";

export interface IAboutProps {}

// todo: make team information interface

export function About(props: IAboutProps) {
  const teamInformation = [];
  return (
    <div className="container">
      <Header as="h1" textAlign="center">
        <Header.Content>Real Time Occupancy Management</Header.Content>
        <Header.Subheader color="blue">About Us</Header.Subheader>
      </Header>
    </div>
  );
}
