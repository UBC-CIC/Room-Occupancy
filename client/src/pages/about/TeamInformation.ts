import Woody from "../../assets/images/team/woody.png";
import Colin from "../../assets/images/team/colin.jpg";
import Steve from "../../assets/images/team/steve.jpg";
import Cassiel from "../../assets/images/team/cassiel.png";
import Danny from "../../assets/images/team/danny.png";

interface ITeamInformation {
  name: string;
  title: string;
  description: string;
  image: string;
  url: string;
}

export const teamInformation: ITeamInformation[] = [
  {
    name: "Colin Pereira",
    title: "Electrical Engineer",
    description:
      "Interested in web development. Strong believer of success through collaboration!",
    image: Colin,
    url: "https://www.linkedin.com/in/colinpereira/",
  },
  {
    name: "Cassiel Jung",
    title: "Computer Engineer",
    description: "Enjoys working with software.",
    image: Cassiel,
    url: "https://www.linkedin.com/in/cassieljung-ml-sw/",
  },
  {
    name: "Woody Chang",
    title: "Electrical Engineer",
    description: "Enjoys working with software.",
    image: Woody,
    url: "https://www.linkedin.com/in/woody-chang-119117158/",
  },
  {
    name: "Danny Song",
    title: "Electrical Engineer",
    description: "Enjoys working with software.",
    image: Danny,
    url: "https://www.linkedin.com/in/minwo02001/",
  },
  {
    name: "Steve He",
    title: "Computer Engineer",
    description: "Enjoys working with software.",
    image: Steve,
    url: "https://www.linkedin.com/in/he-steve/",
  },
];
