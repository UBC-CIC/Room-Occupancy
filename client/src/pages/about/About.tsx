import * as React from "react";
import { Header } from "semantic-ui-react";
import "../../App.css";
import { Cards } from "../../components/Cards";
import { teamInformation } from "./TeamInformation";

interface IAboutProps {}

export function About(props: IAboutProps) {
  return (
    <div className="container">
      <Header color="blue" as="h1" textAlign="center">
        <Header.Content>Our Team</Header.Content>
        <Header.Subheader color="blue">
          Our team comprises of five engineering students within the department
          of Electrical and Computer Engineering at The University of British
          Columbia
        </Header.Subheader>
      </Header>
      <div className="flexContainer">
        {teamInformation.map((member) => {
          return (
            <div className="flexItem" key={member.name}>
              <Cards
                imageURL={member.image}
                header={member.name}
                subheader={member.title}
                description={member.description}
                subDescription={null}
                icon="linkedin"
                url={member.url}
              />
            </div>
          );
        })}
      </div>
    </div>
  );
}
