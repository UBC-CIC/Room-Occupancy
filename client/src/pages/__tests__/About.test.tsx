import * as React from "react";
import { render } from "@testing-library/react";
import { About } from "../about/About";

describe("<About />", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<About />);

    expect(asFragment()).toMatchSnapshot();
  });
});
