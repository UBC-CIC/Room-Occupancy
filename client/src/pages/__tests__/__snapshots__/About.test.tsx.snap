// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1
      class="ui blue center aligned header"
    >
      <div
        class="content"
      >
        Our Team
      </div>
      <div
        class="sub header"
        color="blue"
      >
        Our team comprises of five engineering students within the department of Electrical and Computer Engineering at The University of British Columbia
      </div>
    </h1>
    <div
      class="flexContainer"
    >
      <div
        class="flexItem"
      >
        <div
          class="ui card"
        >
          <div
            class="image"
          >
            <img
              src="test-file-stub"
            />
          </div>
          <div
            class="content"
          >
            <div
              class="header"
            >
              Colin Pereira
            </div>
            <div
              class="meta"
            >
              <span
                class="date"
              >
                Electrical Engineer
              </span>
            </div>
            <div
              class="description"
            >
              Interested in web development. Strong believer of success through collaboration!
            </div>
          </div>
          <div
            class="extra content"
          >
            <a
              href="https://www.linkedin.com/in/colinpereira/"
              target="_blank"
            >
              <i
                aria-hidden="true"
                class="linkedin icon"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        class="flexItem"
      >
        <div
          class="ui card"
        >
          <div
            class="image"
          >
            <img
              src="test-file-stub"
            />
          </div>
          <div
            class="content"
          >
            <div
              class="header"
            >
              Cassiel Jung
            </div>
            <div
              class="meta"
            >
              <span
                class="date"
              >
                Computer Engineer
              </span>
            </div>
            <div
              class="description"
            >
              Enjoys working with software.
            </div>
          </div>
          <div
            class="extra content"
          >
            <a
              href="https://www.linkedin.com/in/cassieljung-ml-sw/"
              target="_blank"
            >
              <i
                aria-hidden="true"
                class="linkedin icon"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        class="flexItem"
      >
        <div
          class="ui card"
        >
          <div
            class="image"
          >
            <img
              src="test-file-stub"
            />
          </div>
          <div
            class="content"
          >
            <div
              class="header"
            >
              Woody Chang
            </div>
            <div
              class="meta"
            >
              <span
                class="date"
              >
                Electrical Engineer
              </span>
            </div>
            <div
              class="description"
            >
              Enjoys working with software.
            </div>
          </div>
          <div
            class="extra content"
          >
            <a
              href="https://www.linkedin.com/in/woody-chang-119117158/"
              target="_blank"
            >
              <i
                aria-hidden="true"
                class="linkedin icon"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        class="flexItem"
      >
        <div
          class="ui card"
        >
          <div
            class="image"
          >
            <img
              src="test-file-stub"
            />
          </div>
          <div
            class="content"
          >
            <div
              class="header"
            >
              Danny Song
            </div>
            <div
              class="meta"
            >
              <span
                class="date"
              >
                Electrical Engineer
              </span>
            </div>
            <div
              class="description"
            >
              Enjoys working with software.
            </div>
          </div>
          <div
            class="extra content"
          >
            <a
              href="https://www.linkedin.com/in/minwo02001/"
              target="_blank"
            >
              <i
                aria-hidden="true"
                class="linkedin icon"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        class="flexItem"
      >
        <div
          class="ui card"
        >
          <div
            class="image"
          >
            <img
              src="test-file-stub"
            />
          </div>
          <div
            class="content"
          >
            <div
              class="header"
            >
              Steve He
            </div>
            <div
              class="meta"
            >
              <span
                class="date"
              >
                Computer Engineer
              </span>
            </div>
            <div
              class="description"
            >
              Enjoys working with software.
            </div>
          </div>
          <div
            class="extra content"
          >
            <a
              href="https://www.linkedin.com/in/he-steve/"
              target="_blank"
            >
              <i
                aria-hidden="true"
                class="linkedin icon"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
